x-pack/plugins/task_manager/server/task_scheduling.mock.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { TaskScheduling } from './task_scheduling';
+
+const createTaskSchedulingMock = () => {
+  return ({
+    ensureScheduled: jest.fn(),
+    schedule: jest.fn(),
+    runNow: jest.fn(),
+  } as unknown) as jest.Mocked<TaskScheduling>;
```

Comment:
Converting to unknown will allow the object above to become a partial of what TaskScheduling is. Doing something like the following would solve the concern:

```
const mocked: jest.Mocked<TaskScheduling > = {
  ensureScheduled: jest.fn(),
  schedule: jest.fn(),
  runNow: jest.fn(),
};
return mocked;
```

Reply:
Nope, sadly this dosn't work because of the `private` fields.
If you try what you posted above you get this error:
```
Type '{ ensureScheduled: jest.Mock<any, any>; schedule: jest.Mock<any, any>; runNow: jest.Mock<any, any>; }' is not assignable to type 'Mocked<TaskScheduling>'.
  Type '{ ensureScheduled: Mock<any, any>; schedule: Mock<any, any>; runNow: Mock<any, any>; }' is missing the following properties from type 'TaskScheduling': store, taskPollingLifecycle, logger, middleware, awaitTaskRunResultts(2322)
```

The only way I've found to get around this is the cast to `unknown`.
In some places we use `PublicMethodsOf<>` to get around it but then it complains when you try to pass in the actual thing 🤦 

Have you found any other solution to this?

Reply:
`PublicMethodsOf<>` is what I've seen solve the problem elsewhere. I'm surprised it would complain about passing the actual thing in some places. I may spike something to see what's up.